aws-s3-transfer-manager/src/operation/upload_objects.rs

```diff
@@ -24,14 +32,56 @@ pub(crate) struct UploadObjects;
 impl UploadObjects {
     /// Execute a single `UploadObjects` transfer operation
     pub(crate) async fn orchestrate(
-        _handle: Arc<crate::client::Handle>,
-        _input: UploadObjectsInput,
-    ) -> Result<UploadObjectsHandle, UploadObjectsError> {
-        unimplemented!()
+        handle: Arc<crate::client::Handle>,
+        input: UploadObjectsInput,
+    ) -> Result<UploadObjectsHandle, crate::error::Error> {
+        //  validate existence of source and return error if it's not a directory
+        let source = input.source().expect("source set");
+        validate_target_is_dir(source).await?;
+
+        let concurrency = handle.num_workers();
+        let ctx = UploadObjectsContext::new(handle.clone(), input);
+
+        // spawn all work into the same JoinSet such that when the set is dropped all tasks are cancelled.
+        let mut tasks = JoinSet::new();
+        let (work_tx, work_rx) = async_channel::bounded(concurrency);
```

Comment:
trivial/debatable: 
```suggestion
        let (list_directory_tx, list_directory_rx) = async_channel::bounded(concurrency);
```

Reply:
I actually prefer new names. They are explicit and help build mental model as I read code. Updated in 74af206.